nrf52-hal-common/src/uarte.rs

```diff
@@ -145,6 +145,10 @@ impl<T> Uarte<T> where T: UarteExt {
         // Reset the event, otherwise it will always read `1` from now on.
         self.0.events_endtx.write(|w| w);
 
+        // Conservative compiler fence to prevent optimizations that do not
+        // take in to account DMA
+        compiler_fence(AcqRel);
```

Comment:
The comment I made in `spim.rs` about a (possibly) missing compiler fence at the method entry applies here too.